Documentation/metrics.md

```diff
@@ -0,0 +1,26 @@
+Metrics
+=======
+
+cloud-credential-operator (CCO) exposes metrics for integration with Prometheus. Metrics are not persistent. Restarting CCO will reset metrics.
+
+The metrics are exposed via HTTP on port 2112 at `/metrics` (eg. http://localhost:2112/metrics).
+
+The metrics names have a `cco` prefix.
+
+### Server
+
+These metrics describe the status of CCO. 
+
+All these metrics are prefixed with `cco_`
+
+| Name                            | Description                                        | Type      |
+|---------------------------------|----------------------------------------------------|-----------|
+| credentials_requests            | Count of credentials requests                      | Gauge     |
+| controller_reconcile_seconds    | Controller execution time in reconcile loop.       | Histogram |
+| credentials_requests_conditions | Credentials requests with active conditions.       | Gauge     |
+
```

Comment:
It might also be good to call out the controller runtime metrics as well, and show some examples of what we get for free. (controller_runtime_reconcile_total most notably)